imports/lib/models/validateSchema.ts

```diff
@@ -0,0 +1,65 @@
+import { z } from 'zod';
+
+export default function validateSchema<
+  T extends z.ZodFirstPartySchemaTypes
+>(schema: T, path: string[] = []) {
+  const { _def: def } = schema;
+
+  switch (def.typeName) {
+    case z.ZodFirstPartyTypeKind.ZodObject:
+      Object.entries(def.shape())
+        .forEach(([key, field]) => validateSchema(field as z.ZodTypeAny, [...path, key]));
+      validateSchema(def.catchall);
+      break;
+    case z.ZodFirstPartyTypeKind.ZodArray:
+      validateSchema(def.type, [...path, '[]']);
+      break;
+    case z.ZodFirstPartyTypeKind.ZodUnion:
+    case z.ZodFirstPartyTypeKind.ZodDiscriminatedUnion:
+      def.options.forEach((option: z.ZodTypeAny) => validateSchema(option));
+      break;
+    case z.ZodFirstPartyTypeKind.ZodIntersection:
+      validateSchema(def.left);
+      validateSchema(def.right);
+      break;
+    case z.ZodFirstPartyTypeKind.ZodTuple:
+      def.items.forEach((item: z.ZodTypeAny, idx: number) => {
+        return validateSchema(item, [...path, idx.toString()]);
+      });
+      break;
+    case z.ZodFirstPartyTypeKind.ZodRecord:
+      validateSchema(def.valueType, [...path, '[]']);
+      break;
+
+    case z.ZodFirstPartyTypeKind.ZodDefault:
+    case z.ZodFirstPartyTypeKind.ZodNullable:
+    case z.ZodFirstPartyTypeKind.ZodOptional:
+      validateSchema(def.innerType);
+      break;
+    case z.ZodFirstPartyTypeKind.ZodEffects:
+      validateSchema(def.schema);
+      break;
+
+    case z.ZodFirstPartyTypeKind.ZodEnum:
+    case z.ZodFirstPartyTypeKind.ZodNativeEnum:
+    case z.ZodFirstPartyTypeKind.ZodLiteral:
+    case z.ZodFirstPartyTypeKind.ZodNumber:
+    case z.ZodFirstPartyTypeKind.ZodDate:
+    case z.ZodFirstPartyTypeKind.ZodBoolean:
+    case z.ZodFirstPartyTypeKind.ZodNever:
+      // No validation needed
+      break;
+
+    case z.ZodFirstPartyTypeKind.ZodString: {
+      // String fields must not accept empty strings
+      const result = schema.safeParse('');
+      if (result.success) {
+        throw new Error(`String fields must not accept empty strings (${path.join('.')})`);
```

Comment:
This exists to help keep us from writing a schema with a string field that permits empty strings, yes?  I guess the need for the possibly-empty string in `ChatMessage` implies that this validation needs some way to permit an exception?